providers/openlineage/src/airflow/providers/openlineage/plugins/adapter.py

```diff
@@ -157,15 +158,30 @@ def emit(self, event: RunEvent):
         transport_type = f"{self._client.transport.kind}".lower()
 
         try:
-            with ExitStack() as stack:
-                stack.enter_context(Stats.timer(f"ol.emit.attempts.{event_type}.{transport_type}"))
-                stack.enter_context(Stats.timer("ol.emit.attempts"))
-                self._client.emit(redacted_event)
-                self.log.info(
-                    "Successfully emitted OpenLineage `%s` event of id `%s`",
-                    event_type.upper(),
-                    event.run.runId,
-                )
+            if AIRFLOW_V_3_1_PLUS:
+                from airflow.metrics.dual_stats_manager import DualStatsManager
+
+                # If enabled on the config, publish metrics twice,
+                # once with backward compatible name, and then with tags.
+                with DualStatsManager.timer(
+                    f"ol.emit.attempts.{event_type}.{transport_type}", "ol.emit.attempts"
+                ):
+                    self._client.emit(redacted_event)
+                    self.log.info(
+                        "Successfully emitted OpenLineage `%s` event of id `%s`",
+                        event_type.upper(),
+                        event.run.runId,
+                    )
```

Comment:
Rather than duplicating the code here, I think it would be better to use the exist stack and have this as something like

```python
            with ExitStack() as stack:
                if AIRFLOW_V_3_1_PLUS:
                    stack.enter_context(DualStatsManager)
                else:
                    stack.enter_context(Stats.timer(...))
                    stack.enter_context(Stats.timer(...))

                self._client.emit(redacted_event)
                self.log.info(
                    "Successfully emitted OpenLineage `%s` event of id `%s`",
                    event_type.upper(),
                    event.run.runId,
                )
```